lib/parser.ex

```diff
@@ -0,0 +1,50 @@
+defmodule DateTimeParser.Parser do
+  @moduledoc false
+  alias DateTimeParser.{Epoch, Serial}
+
+  @epoch_regex ~r|\A(?<seconds>\d{10,11})(?:\.(?<subseconds>\d{1,10}))?\z|
+  @serial_regex ~r|\A-?\d{1,5}(?:\.\d{1,10})?\z|
+  @default_parsers [:epoch, :serial, :tokenizer]
+
+  def get_parser(string, opts) do
+    opts = Keyword.merge([tokenizer: get_tokenizer(opts[:context], string)], opts)
+
+    opts
+    |> Keyword.get(:parsers, Application.get_env(DateTimeParser, :parsers, @default_parsers))
```

Comment:
This is a nice way to have the various ways to set the option fallback on one-another